src/main/java/org/folio/edge/oaipmh/OaiPmhHandler.java

```diff
@@ -71,53 +76,80 @@ protected void handle(RoutingContext ctx) {
     HttpServerRequest request = ctx.request();
     log.debug("Client request: {} {}", request.method(), request.absoluteURI());
     log.debug("Client request parameters: " + request.params());
-    log.debug("Client request headers: " + request.headers());
+    log.debug("Client request headers: " + Iterables.toString(request.headers()));
 
     if (!supportedAcceptHeaders(request)) {
       notAcceptableResponse(ctx, request);
       return;
     }
 
-    getOkapiClient(ctx, (okapiClient, param) -> configurationService.getEnableOaiServiceConfigSetting(okapiClient)
-      .setHandler(oaiPmhEnabled -> {
-        if (!oaiPmhEnabled.result()) {
-          serviceUnavailableResponse(ctx);
-          return;
-        }
-        configurationService.associateErrorsWith200Status(okapiClient)
-          .setHandler(responseStatusShouldBe200 -> {
-            ctx.put(ERRORS_PROCESSING_KEY, responseStatusShouldBe200.result());
-            Verb verb = Verb.fromName(request.getParam(VERB));
-            if (verb == null) {
-              badRequest(ctx, String.format("Bad verb. Verb '%s' is not implemented", request.getParam(VERB)), null, BAD_VERB);
-              return;
-            }
-            List<OAIPMHerrorType> errors = verb.validate(ctx);
-            if (!errors.isEmpty()) {
-              badRequest(ctx, verb.toString(), errors);
-              return;
-            }
-            String[] requiredParams;
-            if (verb.getExclusiveParam() != null && request.getParam(verb.getExclusiveParam()) != null) {
-              requiredParams = new String[] { verb.getExclusiveParam() };
-            } else {
-              requiredParams = verb.getRequiredParams()
-                .toArray(new String[0]);
-            }
-            super.handleCommon(ctx, requiredParams, verb.getOptionalParams()
-              .toArray(new String[0]), (client, params) -> {
-                final OaiPmhOkapiClient oaiPmhClient = new OaiPmhOkapiClient(client);
-                oaiPmhClient.call(request.params(), request.headers(), response -> handleProxyResponse(ctx, response),
-                    t -> oaiPmhFailureHandler(ctx, t));
+    getOkapiClient(ctx, (okapiClient, notUsed) ->
+      configurationService.getEnableOaiServiceConfigSetting(okapiClient)
+        .compose(oaiPmhEnabled -> {
+          if (!oaiPmhEnabled) {
+            serviceUnavailableResponse(ctx);
```

Comment:
(optinal) renaming like this 'doServiceUnavailableResponse' will be more readable i think;